app/models/manageiq/providers/redhat/infra_manager/vm.rb

```diff
@@ -5,7 +5,8 @@ class ManageIQ::Providers::Redhat::InfraManager::Vm < ManageIQ::Providers::Infra
   include_concern 'ManageIQ::Providers::Redhat::InfraManager::VmOrTemplateShared'
 
   supports :migrate do
-    unless ext_management_system.supports_migrate?
+    migrate = ext_management_system.supports_migrate? if ext_management_system
+    unless migrate
```

Comment:
I wonder if we should keep the standard unsupported reason from [here](https://github.com/ManageIQ/manageiq/blob/c087b3fc74b9eecf0715d29455ae76cc54854baf/app/models/vm/operations/lifecycle.rb#L10-L12) if the vm is `archived?` which will also double as a check for `ems_id.nil?`
If a VM is archived/orphaned the unsupported reason shouldn't really be that `'RHV API version does not support migrate'`